R/app_ui.R

```diff
@@ -23,14 +23,40 @@ app_ui = function(request) {
       # Input widgets
       shiny::tabPanel(
         title = "Data",
+
+        shiny::fluidRow(
+          shiny::column(12,
+                        htmltools::tags$details(
+                          htmltools::tags$summary("Download Files (click to expand)"),
+
+                          # choose cluster id
+                          cluster_idUI("choice1"),
+
+                          # output options
+                          shiny::tabsetPanel(id = "plot_tabs",
+
+                                             # Tables tab
+                                             tablesUI("table1"),
+
+                                             # Plots tab
+                                             plotsUI("plot1"),
+
+                                             # RDS tab
+                                             rdsUI("rds1")
+
+                          )
+                        )
+                        )
+          ), # end fluid row
+
+        # Bottom row - show tree (static html output from tfpscanner)
```

Comment:
This is good. Could you explain where the expandability of the "Download Files" panel is encoded, please.

Reply:
The `htmltools::tags$details` and `htmltools::tags$summary` functions create the expandable section, with just html code (no javascript required). The `htmltools::tags$summary` defines what is initially displayed as a title/preview, and the rest of the code inside `htmltools::tags$details` defines the content initially hidden.

I've added a comment into this file for future reference as well.